### ISSUE TITLE
Any plan to create some tutorial to introduce the implementation/usage of different loss functions in MXNet?

### ISSUE THREAD

**Issue by lxduan** [dmlc/mxnet#1561]
Like what Caffe does.

**Comment by freesouls** [comment]
in MXNet it is just not that explicit, it is implicit.
For example, in `softmax_output` layer, the data is just softmax after feedforward, the actually loss is calculated in [metric.py](https://github.com/dmlc/mxnet/blob/master/python/mxnet/metric.py), when backwarding the gradient is directly calculated by [SoftmaxGrad](https://github.com/dmlc/mxnet/blob/master/src/operator/softmax_output-inl.h#L109) function(you can also write your own codes in backward), so basically speaking, it is the same as implementing new layers.
